src/main/java/apoc/convert/Convert.java

```diff
@@ -79,12 +88,91 @@ else if (list.getClass().isArray() && !list.getClass().getComponentType().isPrim
         }
         return null;
     }
-
+    
     @SuppressWarnings("unchecked")
+    private <T> List<T> convertToList(Object list, Class<T> type) {
+        List<Object> convertedList = convertToList(list);
+        if (convertedList == null) {
+        	return null;
+        }
+        Stream<T> stream = null;
+    	if (type.isAssignableFrom(Long.class)) {
+    		final Exact exact = new Exact();
+    		stream = (Stream<T>) convertedList.stream().map(e -> {
+    			if (e == null) {
+    				return null;
+    			}
+    			if (e instanceof Number) {
+    				return ((Number)e).longValue();
+    			}
+    			return exact.toInteger(toString(e), 0L, RoundingMode.HALF_UP.toString());
```

Comment:
this is way to expensive. I'd stick with try catch around Long.parseLong(e.toString()) and return null in case of exception.